### PR TITLE
Add project version function validation test

### PR DESCRIPTION
**Pull request by MrAlias**

### CODE REVIEW

File: version_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+// regex taken from https://github.com/Masterminds/semver/tree/v3.1.1
+var versionRegex = regexp.MustCompile(`^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?` +
+	`(-([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?` +
+	`(\+([0-9A-Za-z\-]+(\.[0-9A-Za-z\-]+)*))?$`)
+
+func TestVersionSemver(t *testing.T) {
+	v := splunkotel.Version()
+	assert.NotNil(t, versionRegex.FindStringSubmatch(v), "version is not semver: %s", v)
```
Comment (pellared) [inline version_test.go:32]: Maybe we can use https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/assert#Regexp instead ?

File: version_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"regexp"
+	"testing"
+
+	splunkotel "github.com/signalfx/splunk-otel-go"
+	"github.com/stretchr/testify/assert"
+)
+
+// regex taken from https://github.com/Masterminds/semver/tree/v3.1.1
+var versionRegex = regexp.MustCompile(`^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?` +
```
Comment (pellared) [inline version_test.go:26]: I think we DO NOT want the `v` prefix

[the remark this answers is not in the document]
Comment (MrAlias) [reply version_test.go:26]: not sure there is a substantive difference given the test currently, but sure, will do.

**Review by pellared — APPROVED** [review]
LGTM :shipit:

**Pull request merged**